### DISCUSSION TITLE
Turbulent Couette Flow

### DISCUSSION THREAD

**Discussion by rugric** [xcompact3d/Incompact3d#153] [Q&A] [answered]
Hi everyone! I'd like to simulate the Couette turbulent flow. Starting from the example of the turbulent flat channel, I need to be able to put the top wall in motion.  With regard to the turbulent flat channel, what is done is to assign no-slip conditions (nlcy1=2 and nclyn=2) to both the bottom and top walls, through Dirichlet conditions. What I was thinking of doing is to maintain the same conditions on the bottom, while for the top wall, I would set a non-zero velocity via Dirichlet conditions. I tried to replace ux(i,xsize(2),k)=byxn(i,k)+dpdxyn(i,k) with ux(i,xsize(2),k)=one+dpdxyn(i,k) in the pre_correc subroutine, but the problem persist. If it is possible, I would like to know what I am doing wrong and which part of the code I should modify to achieve the Couette flow.
Thanks

**Comment by slaizet** [comment, marked as answer]
Hi,
I just did a quick test with ux(i,xsize(2),k)=one+dpdxyn(i,k) + initial condition 0.5+0.5*y (instead of 1-y*y) and it seems to be working fine. I believe you would also have to change the flow rate.
Best,
Sylvain

**Reply by slaizet** [reply]
Hi,
As you can see, this is not what we get here. It seems that you did not change the initial condition properly.
Best,
Sylvain

![couette](https://user-images.githubusercontent.com/45965875/221832538-3bac05fd-1462-4064-a53a-b84affca1e00.png)

**Reply by rugric** [reply]
Yes, thank you. My mistake. Sorry for the inconvenience.
